design-proposals/api-review.md

```diff
@@ -0,0 +1,162 @@
+contributing api-facing changes. 
+
+### How to achieve this?
+
+- Have a document to describe the right process for contributors with the following details:
+  - Any API-facing change with design doc can be reviewed by the sig-api for initial feedback.
+  - Any API-facing change PR can be brought up for the discussion in sig-api call
+  - Link to a conventions document for good practices and guidance
+
+## Tools for reviewers
+
+The easiest part about checking for good api-facing change is one that maintains backward compatibility. API objects are
+serialized to and from JSON when clients interact with API server. The testing for serialization could be automated to 
+make sure that APIs continue to be serializable upon upgrades
+
+### Tool to test serialization upon upgrade
+Here is a [demo](https://github.com/alaypatel07/KubeVirt-api-fuzzer) tool that identifies api breakages while reading 
+older objects using newer clients. 
+
+##### Description and Usage
+1. This tool creates JSON and YAML files for all the API exposed by KubeVirt in group-version "kubevirt.io/v1",
+   versioned by the release. The current version is in `HEAD` directory, previous versions are in `release-0.yy` release
+   directory. The following APIs are included, more APIs can be added in the future:
+    ```
+    VirtualMachineInstance
+    VirtualMachineInstanceList
+    VirtualMachineInstanceReplicaSet
+    VirtualMachineInstanceReplicaSetList
+    VirtualMachineInstancePreset
+    VirtualMachineInstancePresetList
+    VirtualMachineInstanceMigration
+    VirtualMachineInstanceMigrationList
+    VirtualMachine
+    VirtualMachineList
+    KubeVirt
+    KubeVirtList
+    ```
+2. Upon any change to API, the json and YAML files will be updated in HEAD directory.
+3. When KubeVirt cuts a new release of the project, the files in HEAD directory will be copied to the release version and
+   future development branch will add a unit test for past two releases:
+    ```
+    $ VERSION=release-0.60
+    $ cp -fr testdata/{HEAD,${VERSION}} 
+    ```
+   
+##### How will it help?
+
+During KubeVirt upgrade, the apiserver is updated last, i.e. for a moment in time until the upgrade rolls out, KubeVirt
+components like virt-handler, virt-controller will have newer client, but apiserver will be serving older objects.
+
+Using this tests, it can be asserted that the current newer clients can roundtrip (serialized and de-serialized) past
+two releases, which will make the upgrade safer.
+
+
+## API Examples
+Tool usage example: https://github.com/alaypatel07/kubevirt-api-fuzzer#usage
+
+## Scalability
+TODO
+
+## Update/Rollback Compatibility
+(does this impact update compatibility and how)
+
+## Functional Testing Approach
+(an overview on the approaches used to functional test this design)
+
+# Implementation Phases
+(How/if this design will get broken up into multiple phases)
```

Comment:
It will be nice to fill this up.
I guess starting the SIG formally has already occurred, starting reviewing PR/s, working on the guidance document and possibly giving a talk to help contributors and reviewers to follow it.

Reply:
added Implementation Phases, PTAL